common/static/common/js/discussion/utils.js

```diff
@@ -353,31 +370,38 @@
             return this.getWmdEditor($content, $local, cls_identifier).refreshPreview();
         };
 
-        DiscussionUtil.processEachMathAndCode = function(text, processor) {
+        /**
+         * Formats math and code chunks
+         * @param safeHtml - post contents in form of safe (escaped and/or stripped) HTML
+         * @param processor - callback to post-process math and code chunks
+         * @returns {*}
+         */
+        DiscussionUtil.processEachMathAndCode = function(safeHtml, processor) {
             var $div, ESCAPED_BACKSLASH, ESCAPED_DOLLAR, RE_DISPLAYMATH, RE_INLINEMATH, cnt, codeArchive, processedText;
             codeArchive = [];
             RE_DISPLAYMATH = /^([^\$]*?)\$\$([^\$]*?)\$\$(.*)$/m;
             RE_INLINEMATH = /^([^\$]*?)\$([^\$]+?)\$(.*)$/m;
             ESCAPED_DOLLAR = '@@ESCAPED_D@@';
             ESCAPED_BACKSLASH = '@@ESCAPED_B@@';
             processedText = "";
-            $div = $("<div>").html(text);
+            // safe-lint: disable=javascript-jquery-html
+            $div = $("<div>").html(safeHtml);
```

Comment:
You should use `ensureHtml()` if safeHtml is an HTML Snippet that was made safe by another `HtmlUtils` call.  It will be a no-op, but ensure that `HtmlUtils` will have been called.  If it isn't, can you turn it into real `safeHtml` before passing it to this function?